File: generator/src/main/java/ch/bailu/gtk/tag/RepositoryTag.java
package ch.bailu.gtk.tag;

public class RepositoryTag extends Tag {
    public RepositoryTag(Tag parent) {
        super(parent);
    }

    private final NamespaceTag namespace = new NamespaceTag(this);

    @Override
    public Tag getChild(String name, String prefix) {
        if ("namespace".equals(name)) {
            return namespace;
        }

        if ("package".equals(name)) {
            return namespace.addPackage(new NamedTag(this));
        }

        if ("c".equals(prefix) && "include".equals(name)) {
            return namespace.addInclude(new NamedTag(this));
        }

        return new IgnoreTag(this);
    }

}

File: generator/src/main/kotlin/ch/bailu/gtk/parser/tag/RepositoryTag.kt
package ch.bailu.gtk.parser.tag

class RepositoryTag(parent: Tag) : TagWithParent(parent) {
    private val namespace = NamespaceTag(this)

    override fun getChild(name: String, prefix: String): TagWithParent {
        if ("namespace" == name) {
            return namespace
        }
        if ("package" == name) {
            return namespace.addPackage(NamedWithDocTag(this))
        }
        if ("c" == prefix && "include" == name) {
            return namespace.addInclude(NamedWithDocTag(this))
        }
        return ignore()
    }
}
